telegram/_business.py

```diff
@@ -453,3 +455,69 @@ def de_json(
         )
 
         return super().de_json(data=data, bot=bot)
+
+    def get_opening_hours_for_day(
+        self, target_date: dt, tzinfo: Optional[zoneinfo.ZoneInfo] = None
+    ) -> Tuple[Tuple[datetime, datetime], ...]:
+        """
+        Get the opening hours for a specific day.
+
+        .. versionadded:: NEXT.VERSION
+
+        Args:
+            target_date (:class:`datetime.date`): The date for which to get the opening hours.
+            tzinfo (:class:`zoneinfo.ZoneInfo`, optional): The timezone to use for opening hours.
+                If :obj:`None`, the time zone of the business is used, i.e. :attr:`time_zone_name`.
+                Defaults to :obj:`None`.
+
+        Returns:
+            Tuple[Tuple[:class:`datetime.datetime`, :class:`datetime.datetime`], ...]:
+                A tuple of tuples containing the opening and closing times for the day.
+        """
+        if tzinfo is None:
+            tzinfo = zoneinfo.ZoneInfo(self.time_zone_name)
+
+        output = []
+
+        for interval in self.opening_hours:
+            opening_time = datetime.combine(target_date, time(*interval.opening_time[1:]), tzinfo)
+            closing_time = datetime.combine(target_date, time(*interval.closing_time[1:]), tzinfo)
```

Comment:
IISC this blindly assumes that the `interval` is valid for the `target_data`. But the business might be closed on that date. This doesn't look right to me. If I'm wrong, kindly explain why this works